src/adapters/mson.es6

```diff
@@ -0,0 +1,201 @@
+/*
+ * Renders refract elements into MSON.
+ */
+
+/*
+ * Indent a piece of multiline text by a number of spaces.
+ * Setting `first` to `true` will also indent the first line.
+ */
+function indent(input, spaces, options={first: false}) {
+   let pre = '';
+   let lines = [];
+
+   for (let _ = 0; _ < spaces; _++) {
+     pre += ' ';
+   }
+
+   for (let line of input.split('\n')) {
+     // Only indent non-blank lines!
+     lines.push(line ? pre + line : line);
+   }
+
+   lines = lines.join('\n').trim();
+
+   if (options.first) {
+     lines = pre + lines;
+   }
+
+   return lines;
+ }
+
+/*
+ * Get type information for an element, such as the element name, whether
+ * it is required, etc. Returns an array of strings.
+ */
+function getTypeAttributes(element, attributes={}) {
+  let typeAttributes = [];
+
+  if (element.element !== 'string' && element.element !== 'dataStructure') {
```

Comment:
This is why `type` or `name` might have been better, but I'm not opening that can of worms again :tongue: